### ISSUE TITLE
Validation Dice and IoU are (validation)-batch-size dependent.

### ISSUE THREAD

**Issue by tabmoo** [qubvel/segmentation_models.pytorch#330]
Assuming deterministic training and validation regimes (be it Tensorflow or Pytorch) and that only 1 class metric is of interest,  averaging over batches inevitably brings about validation batch size dependence. It seems that "per image" option that I encountered in TF repo and also here and there in this Pytorch repo was intended to address this issue. Am I correct?

As I understand, we can only compare models if we use validation batch = 1 or putting one image per GPU in multi-GPU mode.
